crowbar_framework/app/models/service_object.rb

```diff
@@ -925,6 +925,9 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     apply_locks = []
     applying_nodes = []
 
+    # Cache some node attributes to avoid useless node reloads
+    node_attr_cache = {}
```

Comment:
Wouldn't it make more sense for this to live in `NodeObject`?  If we still need to ensure that `find_node_by_name` etc. retain their non-caching behaviour by default then we could introduce new `find_node_by_name_cached` type of methods which wrap around it with the caching behaviour enabled.  +1 for caching in general though ;-)